### PR TITLE
POTENTIAL, first iteration

### PR DESCRIPTION
**Pull request by landswellsong**
What's in:
- Auto-registration of functions in `functions` package and the `utils.functions.func_by_path` utility to locate them in the Python code (#6).
- POTENTIAL data gets imported, currently only EZ series (#19).
- More or less generalised FTP code, will be reused for test data (#53).

Bonus:
- Cube function estimate by 4 known points utility (#25).
- Static file handling gets some love (#39).
- Removed the migration file (#54).
- Wait for the database before starting promis (#38).

### CODE REVIEW

File: promis/backend_api/promis.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
```
Comment (landswellsong) [inline promis/backend_api/promis.py]: @lyssdod please check if this is ok, I seem to have pulled the old file or something.

**Comment by landswellsong** [comment]
I rebased the history not to depend on the `swaggered` branch. Sorry about that.

File: promis/backend_api/models.py [hunk 2]
```diff
@@ -29,6 +50,9 @@ class Space_project(TranslatableModel):
         description = TextField(blank = True)
         )
 
+    # Function that returns functions to check for updates and fetch them
+    data_func = ForeignKey('Function', null = True, related_name = 'par_dt_func')
```
Comment (elpiankova) [inline promis/backend_api/models.py]: Fix related name.

File: promis/util/files.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+#
+# Unless required by applicable law or agreed to in
+# writing, software distributed under the Licence is
+# distributed on an "AS IS" basis,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied.
+# See the Licence for the specific language governing
+# permissions and limitations under the Licence.
+#
+"""Parser functions for the various common file types involved in the project"""
```
Comment (elpiankova) [inline promis/util/files.py]: change name

File: promis/util/functions.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+# Unless required by applicable law or agreed to in
+# writing, software distributed under the Licence is
+# distributed on an "AS IS" basis,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied.
+# See the Licence for the specific language governing
+# permissions and limitations under the Licence.
+#
+import re
+from importlib import import_module
```
Comment (elpiankova) [inline promis/util/functions.py]: Naming

File: promis/util/functions.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+def func_by_path(path):
+    """Returns the function identified by its fully qualified path string"""
+    # Breaking down to components
+    rexp = r"((?:[a-zA-Z_][a-zA-Z0-9_]*\.)*)([a-zA-Z_][a-zA-Z0-9_]*)"
```
Comment (elpiankova) [inline promis/util/functions.py]: Just use `split`?

Comment (elpiankova) [inline promis/util/functions.py]: `str.rsplit("my.module.func", sep=".", maxsplit=1)`

**Pull request merged**